src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -331,7 +361,23 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                     }
                     else
                     {
-                        ApplyResourceSetInNestedProperty(edmProperty, resource, resourceSetWrapper, readContext);
+                        ODataResourceSetWrapper resSetWrapper = resourceSetWrapper;
```

Comment:
@ElizabethOkerio Maybe I'm missing something... What is the reason for introducing the variable `resSetWrapper` instead of just using `resourceSetWrapper`?

Reply:
solved